web-api/src/persistence/postgres/utils/operation/tryGetLock.ts

```diff
@@ -1,15 +1,26 @@
 import { Database } from '@web-api/database-schema';
+import { hashLockId } from '@web-api/persistence/postgres/utils/mutex';
 import { CompiledQuery, Kysely } from 'kysely';
 
-export const tryGetLock = async (
-  db: Kysely<Database>,
-  hashedLockId: number,
-): Promise<boolean> => {
-  const result = await db.executeQuery<{ pgTryAdvisoryLock: boolean }>(
-    CompiledQuery.raw(
-      'SELECT pg_try_advisory_lock($1) AS "pgTryAdvisoryLock"',
-      [hashedLockId],
-    ),
-  );
-  return result.rows[0].pgTryAdvisoryLock;
+// Danger: never use this outside of acquireLock. This should only be used with a specific scoped db connection which is immediately destroyed after being used
+export const tryGetLocks = async ({
+  db,
+  identifiers,
+}: {
+  db: Kysely<Database>;
+  identifiers: string[];
+}): Promise<{ identifier: string; successfullyLocked: boolean }[]> => {
+  const hashedIdentifiers = identifiers.map(id => hashLockId(id));
+  const idMap = new Map(hashedIdentifiers.map((h, i) => [h, identifiers[i]]));
+
+  const sql = hashedIdentifiers
```

Comment:
Ah, clever. I wonder if we can use `unnest` rather than `union all`. That would give us one `SELECT` rather than multiple.

Reply:
But might not be material, seems like timing is already fast.

Reply:
Just tested it, it _is_ marginally faster (~42 ms vs ~36ms for 200 ids). Why not? I'll update it

Reply:
Lol okey doke :)